utest/parsing/test_lexer.py

```diff
@@ -1816,5 +1816,149 @@ def _verify(self, data, expected, test=False):
         assert_tokens(data, expected, data_only=True)
 
 
+class TestContinue(unittest.TestCase):
+
+    def test_in_keyword(self):
+        data = '    CONTINUE'
+        expected = [(T.CONTINUE, 'CONTINUE', 3, 4),
+                    (T.EOS, '', 3, 12)]
+        self._verify(data, expected)
+
+    def test_in_test(self):
+        # This is not valid usage but that's not recognized during lexing.
```

Comment:
It's true using `CONTINUE` and `BREAK` directly under a test isn't valid, but using them directly inside a user keyword isn't valid either. It should be possible to use them only inside `FOR` either directly or inside `IF` and `TRY` structures. This is different to `RETURN` that can be used anywhere inside a user keyword, also directly under them. In other words, this comment applies also to the above test.

Notice that it's possible to use `Continue For Loop` and `Exit For Loop` keywords directly inside user keywords as long as those user keywords are used inside a loop. That's bad design and not something I want to repeat with `CONTINUE` and `BREAK`. Although those keywords will be eventually deprecated and removed, I wouldn't like to change their behavior to avoid backwards incompatible change. We need to see can that be handled nicely when we get further with `CONTINUE/BREAK`.

Notice also that I have an idea how these error situations could be detected already during lexing/parsing. I can look at that after this PR has been merged and update tests then.

Reply:
I am not sure I am following all your points mentioned here however I have modified test little.

Reply:
This ought to be fine. Can be changed later if parser is enhanced to detect `RETURN` and others when used in a wrong place. That's not something this PR needs to care.